publish/views/asset.py

```diff
@@ -53,3 +55,30 @@ class AssetViewSet(NestedViewSetMixin, DetailSerializerMixin, ReadOnlyModelViewS
 
     filter_backends = [filters.DjangoFilterBackend]
     filterset_class = AssetFilter
+
+    @action(detail=False, methods=['GET'])
+    def paths(self, request, **kwargs):
+        """
+        Return the unique files/directories that directly reside under the specified path.
+
+        The specified path must be a folder (must end with a slash).
+        """
+        path_prefix: str = self.request.query_params.get('path_prefix', '/')
+        qs = self.get_queryset().filter(path__startswith=path_prefix).values()
+
+        paths = set()
+        for asset in qs:
+            path_parts = [part for part in asset['path'].split('/') if part]
+            prefix_parts = [part for part in path_prefix.split('/') if part]
+
+            # If the prefix is the exact path of this asset, or the prefix parts aren't a strict
+            # subset of the asset path parts (a folder which doesn't exist), skip this asset
+            if path_prefix == asset['path'] or (set(prefix_parts) - set(path_parts)):
```

Comment:
`set` is unordered, so it seems weird to do this set subtraction operation. I don't understand how that could every be false; `path_parts` is a set of all components of the asset path, which was return from the `path__startswith=path_prefix` query, so it must necessarily contain all elements of `path_prefix`.

Since these are simple lists of strings, I don't see what's wrong with something like:
```
len(prefix_parts) < len(path_parts) and prefix_parts = path_parts[:len(prefix_parts)]
```

Reply:
The set subtraction was to catch paths that are still prefixes of asset paths, but aren't actually full folders. So if the asset path is `/folder/subfolder/asset.nwb`, and the prefix passed was `/folder/sub`, `set(prefix_parts) == {folder, sub}`, where `set(path_parts) == {folder, subfolder, asset.nwb}`. Since that `sub` part isn't in `path_parts`, it would skip this, as it's not valid here.